src/event_handler.rs

```diff
@@ -192,22 +188,125 @@ async fn send_execution_output(app_state: Data<AppState>, mut rx: Receiver<JobRe
+                sleep(Duration::from_millis(10)).await;
+                continue;
+            };
+
+            // Get current gas price for the common chain network and retry otherwise
+            let http_rpc_client = app_state.http_rpc_client.lock().unwrap().clone().unwrap();
+            let gas_price = http_rpc_client.get_gas_price().await;
+            let Ok(gas_price) = gas_price else {
+                eprintln!(
+                    "Failed to get gas price from the rpc for the network: {:?}",
+                    gas_price.unwrap_err()
                 );
+                sleep(Duration::from_millis(10)).await;
+                continue;
+            };
 
-                send_txn(txn).await
-            },
-            should_retry,
-        )
-        .await;
+            // If required retrieve the current nonce from the network and retry otherwise
+            if update_nonce == true {
+                let new_nonce_to_send = http_rpc_client
+                    .get_transaction_count(http_rpc_client.address(), None)
+                    .await;
+                if new_nonce_to_send.is_err() {
+                    eprintln!(
+                        "Failed to fetch current nonce for the gas address ({:?}): {:?}",
+                        http_rpc_client.address(),
+                        new_nonce_to_send.unwrap_err()
+                    );
+                    continue;
+                };
+                nonce_to_send = new_nonce_to_send.unwrap();
+                update_nonce = false;
+            }
 
-        let Ok(_) = txn_result else {
-            eprintln!(
-                "Failed to submit the execution output transaction for job id {}: {:?}",
-                job_output.id,
-                txn_result.unwrap_err()
+            // Update metadata to be used for sending the transaction and send it to the common chain
+            let txn = txn
+                .gas(estimated_gas + GAS_LIMIT_BUFFER)
+                .nonce(nonce_to_send)
+                .gas_price(gas_price);
+            let pending_txn = txn.send().await;
+            let Ok(pending_txn) = pending_txn else {
+                let error_string = format!("{:?}", pending_txn.unwrap_err());
+                eprintln!(
+                    "Failed to send the execution {} transaction for job id {}: {}",
+                    txn_type, job_id, error_string
+                );
+
+                // Retry if 'nonce too low' error encountered
+                if error_string.contains("code: -32000") && error_string.contains("nonce") {
+                    update_nonce = true;
+                    continue;
+                }
+
+                // Retry after a delay if connection failed
+                if error_string.contains("code: -32000") && error_string.contains("connection") {
+                    sleep(Duration::from_millis(100)).await;
+                    continue;
+                }
+
```

Comment:
Add comment here to explore rest errors and their handling.

Reply:
So these are the major errors the executor might face while sending a txn and how they are handled currently :- 
1. "nonce too low" :- fetch current nonce for the gas address from the rpc and retry txn
2. "nonce too high" :- this error is generally not thrown by the network because the high nonce txns are not straight away rejected but are included in the mempool waiting to be executed until all the remaining previous nonces have been included. Also we are not skipping any nonce here because we only increment it by 1 when the txn is sent successfully to the network. Might happen that the owner uses the gas address outside the executor to send txns but in that case we'll encounter 'nonce too low' error and that is handled above.
3. "out of gas" :- this is avoided by estimating the gas required for the txn (required before each retry because the contract state might be changing rapidly) and adding a big enough buffer to it for setting the gas limit 
4. "gas price too low" :- handled by fetching the current gas price for the network and using it while sending txns  
5. "connection" issues :- retry after a delay  
6. Rest errors that might arise in our case like "execution reverted" or "insufficient funds" or "contract execution failed" are all non-retryable

I've handled all these scenarios in the current code, will add a comment mentioning the cases.

Reply:
There is a maximum limit on how many transactions can be in the mempool for an account. If that limit is reached for our gas account then we'll get this error. Also we're only incrementing the next nonce to be used when we send a transaction for that successfully to the network. So if we're encountering nonce too high error then it means that some previous transactions (without any nonce gap) are still stuck in the mempool and the limit has been reached for the gas account. So here we just have 2 options :- wait for the previous transactions to be mined or resend the current transaction with the appropriate nonce (transaction count for the gas account) as a replacement txn. In the current code we are doing the later.

Reply:
Do you think current approach is correct?